### PR TITLE
Snapshot: add user guide and move documentation to a common directory

### PR DESCRIPTION
**Pull request by tsmetana**
Here's update of the snapshot documentation: I put everything to one directory named "doc" and added some basic user guide written by @pospispa .

### CODE REVIEW

File: snapshot/doc/user-guide.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+# Features
+
+* Create snapshot of a `PersistentVolume` bound to a `PersistentVolumeClaim`
+* List existing `VolumeSnapshots`
+* Delete existing `VolumeSnapshot`
+* Create a new `PersistentVolume` from an existing `VolumeSnapshot`
+* Supported `PersistentVolume` [types](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#types-of-persistent-volumes):
+    * Amazon EBS
+    * GCE PD
+    * HostPath
```
Comment (rootfs) [inline snapshot/doc/user-guide.md]: OpenStack Cinder

File: snapshot/doc/user-guide.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+Volume Snapshot Data is automatically created based on the Volume Snapshot. Relationship between Volume Snapshot and Volume Snapshot Data is similar to the relationship between Persistent Volume Claim and Persistent Volume.
+
+Depending on the Persistent Volume type the operation might go through several phases which are reflected by the Volume Snapshot status:
+1. The Volume Snapshot Data is created and is bound to the Volume Snapshot.
+2. The controller starts the snapshot operation: the snapshotted Persistent Volume might need to be frozen and the applications paused.
+3. The storage system finishes creating the snapshot (the snapshot is "cut") and the snapshotted Persistent Volume might return to normal operation. The snapshot itself is not ready yet. The last status condition is of `Created` type with status value "True"
+4. The newly created snapshot is completed and ready to use. The last status condition is of `Ready` type with status value "True"
+
+A Volume Snapshot status can be displayed as shown below:
+```yaml
```
Comment (rootfs) [inline snapshot/doc/user-guide.md]: move to below kubectl

File: snapshot/doc/user-guide.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    conditions:
+    - lastTransitionTime: null
+      message: Snapshot created successfully
+      reason: ""
+      status: "True"
+      type: Ready
+    creationTimestamp: null
+```
+
+## Restoring
```
Comment (rootfs) [inline snapshot/doc/user-guide.md]: dup of snapshot promoter section?

File: snapshot/doc/user-guide.md (hunk 1 above)
Comment (pospispa) [reply snapshot/doc/user-guide.md]: Is OpenStack Cinder already supported for snapshots?

Comment (rootfs) [reply snapshot/doc/user-guide.md]: yes, kudos @xing-yang and @j-griffith

File: snapshot/doc/user-guide.md (hunk 2 above)
Comment (pospispa) [reply snapshot/doc/user-guide.md]: Or split it into two "verbatim" parts:
```
$ kubectl get volumesnapshot -o yaml
```
```yaml
apiVersion: volume-snapshot-data.external-storage.k8s.io/v1
  kind: VolumeSnapshot
...
```

File: snapshot/doc/user-guide.md (hunk 3 above)
Comment (pospispa) [reply snapshot/doc/user-guide.md]: I'm doing my best to look at the snaphotting from a user perspective. That's why I split it into Prerequisites and the actual features.
So, firstly, what user needs before being able to use the snapshotting features, i.e. the prerequisites (existing Storage Class pointing to the Snapshot Promoter is one of them).
Secondly, the features, i.e. creating/restoring/deleting a snapshot.

That's why the Snapshot Promoter subsection is a part of the Prerequisites section and Restoring is a separate section as Creating and Deleting.

Does it make sense?

B.T.W., admin guide, i.e. how to install "all the necessary parts", doesn't exist so far.

File: snapshot/doc/user-guide.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+```
+
+* `persistentVolumeClaimName`: name of the Persistent Volume Claim that is bound to a Persistent Volume. This particular Persistent Volume will be snapshotted.
+
+Volume Snapshot Data is automatically created based on the Volume Snapshot. Relationship between Volume Snapshot and Volume Snapshot Data is similar to the relationship between Persistent Volume Claim and Persistent Volume.
+
+Depending on the Persistent Volume type the operation might go through several phases which are reflected by the Volume Snapshot status:
+1. The Volume Snapshot Data is created and is bound to the Volume Snapshot.
+2. The controller starts the snapshot operation: the snapshotted Persistent Volume might need to be frozen and the applications paused.
+3. The storage system finishes creating the snapshot (the snapshot is "cut") and the snapshotted Persistent Volume might return to normal operation. The snapshot itself is not ready yet. The last status condition is of `Created` type with status value "True"
```
Comment (xing-yang) [inline snapshot/doc/user-guide.md]: In the current code, the status condition should be of "Pending" type with status value "True". 

https://github.com/kubernetes-incubator/external-storage/blob/master/snapshot/pkg/apis/crd/v1/types.go#L58

File: snapshot/doc/user-guide.md (hunk 1 above)
Comment (tsmetana) [reply snapshot/doc/user-guide.md]: This is my fault. I will add Cinder.

File: snapshot/doc/user-guide.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+```yaml
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: snapshot-promoter
+provisioner: volumesnapshot.external-storage.k8s.io/snapshot-promoter
+```
+Such Storage Class is necessary for restoring a Persistent Volume from already created Volume Snapshot and Volume Snapshot Data.
+
+## Creating
```
Comment (rootfs) [inline snapshot/doc/user-guide.md]: nit: Creating Snapshot

File: snapshot/doc/user-guide.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+    conditions:
+    - lastTransitionTime: null
+      message: Snapshot created successfully
+      reason: ""
+      status: "True"
+      type: Ready
+    creationTimestamp: null
+```
+
+## Restoring
```
Comment (rootfs) [inline snapshot/doc/user-guide.md]: nit: Restoring snapshot

File: snapshot/doc/user-guide.md [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+    snapshot.alpha.kubernetes.io/snapshot: snapshot-demo
+spec:
+  storageClassName: snapshot-promoter
+```
+* `annotations`: `snapshot.alpha.kubernetes.io/snapshot`: the name of the Volume Snapshot that will be restored.
+* `storageClassName`: Storage Class created by admin for restoring Volume Snapshots.
+
+A Persistent Volume will be created and bound to the Persistent Volume Claim. The process may take several minutes depending on the Persistent Volume Type.
+
+## Deleting
```
Comment (rootfs) [inline snapshot/doc/user-guide.md]: nit: Deleting snapshot

File: snapshot/doc/user-guide.md (hunk 5 above)
Comment (tsmetana) [reply snapshot/doc/user-guide.md]: OK. Will change those.

**Comment by xing-yang** [comment]
My comments are addressed.  LGTM.

**Comment by tsmetana** [comment]
I have included the changes by @liangxia from the PR#391

File: snapshot/doc/user-guide.md [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+
+* It is the user's responsibility to ensure the data consistency (stop the pod/application, flush caches, freeze the filesystem, ...).
+* In case of error in any of the steps the Volume Snapshot status is appended with an `Error` condition.
+
+A Volume Snapshot status can be displayed as shown below:
+```sh
+$ kubectl get volumesnapshot -o yaml
+```
+```yaml
+apiVersion: volume-snapshot-data.external-storage.k8s.io/v1
```
Comment (rootfs) [inline snapshot/doc/user-guide.md]: ditto

File: snapshot/doc/user-guide.md [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+  metadata:
+    clusterName: ""
+    creationTimestamp: 2017-09-19T13:58:28Z
+    generation: 0
+    labels:
+      Timestamp: "1505829508178510973"
+    name: snapshot-demo
+    namespace: default
+    resourceVersion: "780"
+    selfLink: /apis/volume-snapshot-data.external-storage.k8s.io/v1/namespaces/default/volumesnapshots/snapshot-demo
```
Comment (rootfs) [inline snapshot/doc/user-guide.md]: ditto

File: snapshot/doc/user-guide.md [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+metadata:
+  name: snapshot-promoter
+provisioner: volumesnapshot.external-storage.k8s.io/snapshot-promoter
+```
+Such Storage Class is necessary for restoring a Persistent Volume from already created Volume Snapshot and Volume Snapshot Data.
+
+## Creating Snapshot
+Each `VolumeSnapshot` contains a spec and status, which is the specification and status of the Volume Snapshot.
+```yaml
+apiVersion: volume-snapshot-data.external-storage.k8s.io/v1
```
[the remark this answers is not in the document]
Comment (tsmetana) [reply snapshot/doc/user-guide.md]: All right. I'll fix it.

File: snapshot/doc/user-guide.md (hunk 9 above)
Comment (tsmetana) [reply snapshot/doc/user-guide.md]: ... and this too.

File: snapshot/doc/user-guide.md (hunk 8 above)
Comment (tsmetana) [reply snapshot/doc/user-guide.md]: ...and this

**Pull request merged**